### DISCUSSION TITLE
Dynamic import with data attributes

### DISCUSSION THREAD

**Discussion by FlorisdeG** [vercel/next.js#33065] [Help]
We use the code bellow to import our `svg`'s, but currently the importing waits until the script of `cookiebot` is loaded, even though it doesn't require a cookie check. There is a workaround, which is to add a `data-ignoreconsent` data attribute to the script. This way the script will not wait until the `cookiebot` is loaded and we can load the svg's earlier which improves the performance. Is there a way to add this data tag to a dynamic import like the one below?

```
export const ArrowRight = dynamic<{ className?: string }>(
  async () => await import('@icons/arrow.svg'),
  {
    loading: () => <span />,
    ssr: false
  }
)
```

**Comment by cholazzzb** [comment]
I have the same issue, I need to specify the data tag to dynamic import. But it seems that currently next.js does not support this feature? CMIIW

**Reply by cholazzzb** [reply]
@FlorisdeG 
In my case, I solve the issue like this:
```typescript
const DatasheetContainer = dynamic(
  () => import('@/DatasheetContainer'),
  {ssr: false},
) as <DataType extends Record<string, unknown>>({ prop1, prop2 }: DatasheetContainerProps<DataType>) => JSX.Element
```

and call it like this:
```typescript
<DatasheetContainer<YourDataType>
/>
```
